crypto/poseidon/poseidon.go

```diff
@@ -110,57 +110,42 @@ func Hash(inpBI []*big.Int, width int) (*big.Int, error) {
 
 // Hash using possible sponge specs specified by width (rate from 1 to 15), the size of input is applied as capacity
 // (notice we do not include width in the capacity )
-func HashWithCap(inpBI []*big.Int, width int, cap int64) (*big.Int, error) {
+func HashWithCap(inpBI []*big.Int, width int, nBytes int64) (*big.Int, error) {
 	if width < 2 {
 		return nil, fmt.Errorf("width must be ranged from 2 to 16")
 	}
 	if width-2 > len(NROUNDSP) {
 		return nil, fmt.Errorf("invalid inputs width %d, max %d", width, len(NROUNDSP)+1) //nolint:gomnd,lll
 	}
 
-	capflag := ff.NewElement().SetBigInt(big.NewInt(cap))
+	// capflag = nBytes * 2^64
+	pow64 := big.NewInt(1)
+	pow64.Lsh(pow64, 64)
+	capflag := ff.NewElement().SetBigInt(big.NewInt(nBytes))
+	capflag.Mul(capflag, ff.NewElement().SetBigInt(pow64))
 
+	// initialize the state
 	state := make([]*ff.Element, width)
 	state[0] = capflag
 	for i := 1; i < width; i++ {
 		state[i] = zero()
 	}
 
 	rate := width - 1
-	var absorb []*ff.Element
-	for len(inpBI) > 0 {
-		// sponge for fully absorb
-		if l := len(absorb); l != 0 {
-			//sanity check
-			if l != rate {
-				panic("unexpected absorption size")
-			}
-
-			for i, elm := range absorb {
-				state[i+1].Add(state[i+1], elm)
-			}
-			state = permute(state, width)
+	i := 0
```

Comment:
I simplify the computation logic here. Could you review this part? @noel2004 @Thegaram

Reply:
Have looked through the renewed computation logic, think it should be all right.